### PR TITLE
chore: added a useNavigation hook

### PR DESCRIPTION
**Pull request by nlewis84**
We found that navigation was not being passed into NodeSingleConnected. Instead of spreading props again, we decided to use the useNavigation hook within NodeFeaturesConnected to solve the problem and be more verbose with our props.

### CODE REVIEW

**Review by redreceipt — DISMISSED** [review]

**Comment by redreceipt** [comment]
I dismissed my approval. Fix the tests and I'll approve again. Just wanna see the extent of the snap changes.

**Review by conrad-vanl — DISMISSED** [review]
I'd suggest considering to keep the prop spreading. Since this is a `named` component, the prop spreading can actually be very helpful to send overrides to the underlying `Component`. I know it's a bit of react anti-pattern, but removing the prop-spreading is technically a breaking change and since we have a lot of apps floating around, it is hard to test if it actually breaks any of them (there could be apps relying on the prop spreading today) 😬

**Comment by redreceipt** [comment]
@conrad-vanl normally I would agree with you but the underlying component `FeaturesFeedConnected` is also named. So the override is easily moved.

**Comment by conrad-vanl** [comment]
dismiss my review and approve then, I'm just not seeing that it's strictly needed for this PR and is a breaking change

**Comment by nlewis84** [comment]
This article was useful in understanding why the tests were failing. 

https://spin.atomicobject.com/2021/02/24/react-navigation-5-unit-testing-components/

**Comment by nlewis84** [comment]
Ergh, `yarn test -u` is not changing the snaps anymore, but the test is still failing in the CI here (but not locally).

File: packages/apollos-ui-connected/jest.setup.js [hunk 1]
```diff
@@ -37,3 +37,14 @@ jest.mock('@apollosproject/ui-analytics', () => ({
   TrackEventWhenLoaded: () => null,
   withTrackOnPress: (Component) => (props) => <Component {...props} />,
 }));
+
+jest.mock('@react-navigation/native', () => {
```
[the remark this answers is not in the document]
Comment (nlewis84) [reply packages/apollos-ui-connected/jest.setup.js:41]: Without this the tests were failing due to the navigation prop being called when it did not exist one level up. HMW approach this differently than this `.mock`?

[the remark this answers is not in the document]
Comment (redreceipt) [reply packages/apollos-ui-connected/jest.setup.js:41]: we wrap the component in our `WithReactNavigator` test utility. but that's ok. I'll approve this, it doesn't hurt.

**Review by redreceipt — APPROVED** [review]

**Pull request merged**